api-server/src/main/java/io/enmasse/api/server/RequestLogger.java

```diff
@@ -4,19 +4,56 @@
  */
 package io.enmasse.api.server;
 
+import io.enmasse.metrics.api.Metric;
+import io.enmasse.metrics.api.MetricType;
+import io.enmasse.metrics.api.MetricValue;
+import io.enmasse.metrics.api.Metrics;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.container.ContainerResponseContext;
 import javax.ws.rs.container.ContainerResponseFilter;
-import java.util.concurrent.TimeUnit;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
 
 public class RequestLogger implements ContainerRequestFilter, ContainerResponseFilter {
     private static final Logger log = LoggerFactory.getLogger(RequestLogger.class);
     private static final String PROP_NAME = "request.start";
 
+    private long numTotal = 0;
+    private long num4xx = 0;
+    private long num5xx = 0;
+
+    // Dont update metrics on every request to ensure this is a light-weight operation. During normal operation
+    // probes and other requests will ensure this is usually invoked within the interval.
+    private final Clock clock;
+    private final Duration metricUpdateInterval = Duration.ofSeconds(5);
+    private Instant lastMetricUpdate = Instant.MIN;
+    private final Metric totalRequestsMetric = new Metric(
+            "api_server_requests_total",
+            "The total number of HTTP requests",
+            MetricType.counter);
+
+    private final Metric code4xxRequestsMetrics = new Metric(
+            "api_server_requests_4xx_total",
```

Comment:
I don't have production experience of Prometheus so I'd happily defer to other how do, but I wonder if status code should be normalised as a label so that the Prometheus user can do this: http_requests_total{status!~"4.."} to find all the client errors.

I'm being guided by the docs:

https://prometheus.io/docs/practices/naming/#labels
https://prometheus.io/docs/prometheus/latest/querying/examples/

Reply:
Good point, I agree labels is the appropriate thing to use. I will update the pr